internal/cri/server/sandbox_stats_linux.go

```diff
@@ -38,7 +38,8 @@ func (c *criService) podSandboxStats(
 	meta := sandbox.Metadata
 
 	if sandbox.Status.Get().State != sandboxstore.StateReady {
-		return nil, fmt.Errorf("failed to get pod sandbox stats since sandbox container %q is not in ready state: %w", meta.ID, errdefs.ErrUnavailable)
+		log.G(ctx).WithField("podsandboxid", sandbox.ID).Debugf("failed to get pod sandbox stats since sandbox container %q is not in ready state", meta.ID)
```

Comment:
is it client side issue? I mean that we use fmt.Errorf(%w) to mark root cause as ErrUnavailable. The client side should unwrap error.

Reply:
@fuweid please see where this is called from within containerd. (wrote a long PR text to describe why this is needed)

Reply:
Sorry, I didn't follow the description.

https://github.com/containerd/containerd/blob/9a2b85561acf7c4f970f947770ee5152358274c1/internal/cri/server/sandbox_stats_list.go#L39-L49

Checked the main branch and we did expect err to unavailable. 

And the error looks weird because it doesn't have unavailable information. Do you know how to check what containerd version used in CI?

```
: failed to get pod sandbox stats since sandbox container \"c821a1cfa253eca57db68a1ae85184359d1422c7595982e32f47addeb8d8a78e\" is not in ready state"
```

https://github.com/containerd/containerd/commit/f914edf4f68ce68a1ac02e754e8bf1fe6ced55ad

Reply:
containerd://1.7.13

Reply:
@fuweid yes originally i saw it in 1.7.13, but inspecting the code in main i see the issue can be here as well. right?

Reply:
I'd like to fix it first here and then cherry pick back to 1.7 branch

Reply:
I don't think so. main doesn't have the issue you mentioned. please provide the log. thanks

Reply:
For the ErrUnabailable error, we did use `errors.Is` to check. That's why the issue has been fixed.

Based on your description, the error is like 

```
no running task found
```

I think we should skip task not found error instead of unavailable.